jerry-core/include/jerryscript-port.h

```diff
@@ -16,233 +16,301 @@
+ * @param buffer_p: buffer returned by jerry_port_line_read
  */
-double jerry_port_get_current_time (void);
+void jerry_port_line_free (jerry_char_t *buffer_p);
 
 /**
- * Get the current context of the engine. Each port should provide its own
- * implementation of this interface.
+ * jerry-port-io @}
+ */
+
+/**
+ * @defgroup jerry-port-fd Filesystem API
+ * @{
+ */
+
+/**
+ * Canonicalize a file path.
  *
- * Note:
- *      This port function is called by jerry-core when
- *      JERRY_EXTERNAL_CONTEXT is enabled. Otherwise this function is not
- *      used.
+ * If possible, the implementation should resolve symbolic links and other directory references found in the input path,
+ * and create a fully canonicalized file path as the result.
  *
- * @return the pointer to the engine context.
+ * The function may return with NULL in case an error is encountered, in which case the calling operation will not
+ * proceed.
+ *
+ * The implementation should allocate storage for the result path as necessary. Non-NULL return values will be passed
+ * to `jerry_port_path_free` when the result is no longer needed by the caller, which can be used to finalize
+ * dynamically allocated buffers.
+ *
+ * NOTE: The implementation must not return directly with the input, as the input buffer is released after the call.
+ *
+ * @param path_p: zero-terminated string containing the input path
+ * @param path_size: size of the input path string in bytes, excluding terminating zero
+ *
+ * @return buffer with the normalized path if the operation is successful,
+ *         NULL otherwise
  */
-struct jerry_context_t *jerry_port_get_current_context (void);
+jerry_char_t *jerry_port_path_normalize (const jerry_char_t *path_p, jerry_size_t path_size);
 
 /**
- * Makes the process sleep for a given time.
- *
- * Note:
- *      This port function is called by jerry-core when JERRY_DEBUGGER is
- *      enabled (set to 1). Otherwise this function is not used.
+ * Free a path buffer returned by jerry_port_path_normalize.
  *
- * @param sleep_time milliseconds to sleep.
+ * @param path_p: the path buffer to free
  */
-void jerry_port_sleep (uint32_t sleep_time);
+void jerry_port_path_free (jerry_char_t *path_p);
 
 /**
- * Print a single character.
+ * Get the offset of the basename component in the input path.
+ *
+ * The implementation should return the offset of the first character after the last path separator found in the path.
+ * This is used by the caller to split the path into a directory name and a file name.
  *
- * Note:
- *      This port function is here so the jerry-ext components would have
- *      a common way to print out information.
- *      If possible do not use from the jerry-core.
+ * @param path_p: input zero-terminated path string
  *
- * @param c the character to print.
+ * @return offset of the basename component in the input path
  */
-void jerry_port_print_char (char c);
+jerry_size_t jerry_port_path_base (const jerry_char_t *path_p);
```

Comment:
Add length parameter to jerry_port_path_base

Reply:
> The doc states that `param path_p` must be zero terminated string.

Still we can add the path_size paramter for consistence with `jerry_port_path_normalize`, and the first parameter of `jerry_port_path_normalize` also are zero terminated, the port user can choose what-ever they want, use `zero` terminating or string size directly.

Reply:
See my comment [here](https://github.com/jerryscript-project/jerryscript/pull/4907#discussion_r784776452)